src/ws/session.c

```diff
@@ -198,12 +196,50 @@ write_auth_begin (int result_code)
 }
 
 static void
+write_auth_begin (void)
+{
+  authf = open_memstream (&auth_msg, &auth_msg_size);
+  fprintf (authf, "{ ");
+}
+
+static void
 write_auth_end (void)
 {
-  fprintf (authf, " }\n");
+  int r;
+
+  fprintf (authf, "}\n");
+  fflush (authf);
+  fclose (authf);
+
+
+  for (;;)
+    {
+      r = write (AUTH_FD, auth_msg, auth_msg_size);
```

Comment:
Corner case if you send zero bytes write() won't send a packet. Worth an assert() and a comment? Probably not worth switching to send() or sendmsg().